backend/handlers.py

```diff
@@ -178,6 +185,29 @@ def _check_state(self, state):
         cookie_state = self.get_secure_cookie('state').decode('ascii')
         return state == cookie_state
 
+    def _update_to_elixir_account(self, user, from_account_type = "google"):
+        """
+        Takes a previous login session and updates the user in the database
+        """
+        try:
+            _email = self.get_secure_cookie('email').decode('utf-8')
+            _identity_type = self.get_secure_cookie('identity_type').decode('utf-8')
+
+            if _identity_type == from_account_type:
+                db.User.update( name = user["name"],
+                                email = user["email"],
+                                identity = user["sub"],
+                                identity_type = 'elixir'
+                               ).where( db.User.email == _email ).execute()
+        except AttributeError as e:
+            # This will happen whenever we don't have a previous login, so this is fine
+            pass
+        except Exception as e:
```

Comment:
You should use the `self.send_error` method instead of calling `write_error` directly. Also if there's an error the status code should not be `200 Ok`. These are probably `400`.